Seeing JavaScript errors in developer tools on each page

Reported by @snyderp 

<!--
Have you searched for similar issues? We have received a lot of feedback and bug reports that we have closed as duplicates. Before submitting this issue, please visit our community site for common ones: https://community.brave.com/c/common-issues
-->

### Description
When opening the developer tools on any page (the content developer tools), you'll notice a lot of JavaScript errors


### Steps to Reproduce
<!--
Please add a series of steps to reproduce the problem. See https://stackoverflow.com/help/mcve for in depth information on how to create a minimal, complete, and verifiable example.
-->

  1. Visit https://brave.com
  2. Open developer tools
  3. View console


**Actual result:**
<!--
Please add screenshots if needed.
-->
![screen shot 2018-07-31 at 11 23 10 pm](https://user-images.githubusercontent.com/4733304/43504550-bcc82be4-9518-11e8-834b-1ebceb202235.png)

```
Uncaught ReferenceError: require is not defined
    at adInsertion.js:5
pageInformation.js:159 error resolving rule at position #0
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #1
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #2
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #3
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #4
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #5
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #6
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #7
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #8
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #9
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #10
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #11
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #12
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #13
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #14
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #15
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #16
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #17
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
chrome.ipcRenderer.once @ pageInformation.js:159
pageInformation.js:159 error resolving rule at position #18
 Error: invalid expression
    at resolve (pageInformation.js:119)
    at EventEmitter.chrome.ipcRenderer.once (pageInformation.js:157)
    at Object.onceWrapper (VM66 extensions::event_emitter:292)
    at emitTwo (VM66 extensions::event_emitter:105)
    at EventEmitter.emit (VM66 extensions::event_emitter:193)
    at EventEmitter.ipcRenderer.emit (VM65 extensions::ipc_utils:35)
```

**Expected result:**
no JavaScript errors

**Reproduces how often:**
<!--
What percentage of the time does it reproduce?
-->
100%

### Brave Version

**about:brave info:**
<!--
Please open about:brave, copy the version information, and paste it.
-->


**Reproducible on current live release:**
<!--
Is this a problem with the live build? It matters for triage reasons.
-->


### Additional Information
<!--
Any additional information, related issues, extra QA steps, configuration or data that might be necessary to reproduce the issue.
-->
 
Brave: 0.23.73 
V8: 6.8.275.24 
rev: 50bdb6df42550dd14f5636770ec8585aa26e361b 
Muon: 8.0.3 
OS Release: 17.7.0 
Update Channel: Release 
OS Architecture: x64 
OS Platform: macOS 
Node.js: 7.9.0 
Brave Sync: v1.4.2 
libchromiumcontent: 68.0.3440.75

--- a/app/extensions/brave/content/scripts/adInsertion.js
+++ b/app/extensions/brave/content/scripts/adInsertion.js
@@ -2,8 +2,6 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-const crypto = require('brave-crypto')
-
 if (chrome.contentSettings.adInsertion == 'allow') {
   /**
    * Determines the ad size which should be shown
@@ -74,7 +72,11 @@ if (chrome.contentSettings.adInsertion == 'allow') {
     // generate a random segment
     // @todo - replace with renko targeting
     var segments = ['IAB2', 'IAB17', 'IAB14', 'IAB21', 'IAB20']
-    var segment = segments[crypto.random.uniform(segments.length)]
+    // TODO(riastradh): Can't use brave-crypto's random.uniform(n)
+    // here because this is not node.  Use n*Math.random() because
+    // this doesn't seem to be security-sensitive (if itis used at
+    // all?).
+    var segment = segments[Math.floor(segments.length * Math.random())]
     var time_in_segment = new Date().getSeconds()
     var segment_expiration_time = 0 // no expiration
 

